extract-first-url/main.go

```diff
@@ -27,33 +27,28 @@ import (
 
 func main() {
 	flag.Parse()
+
 	action := githubactions.New()
 	internal.DebugEnv(action)
 
-	// open a file
-	file, err := os.Open("./deploy.txt")
-	if err != nil {
-		action.Fatalf("%s", err)
+	if u := extractURL(action, "deploy.txt"); u != "" {
+		action.SetOutput("extracted_url", u)
 	}
-	defer file.Close()
-
-	// get first url form file
-	url := getFirstURLFromFile(file)
+}
 
-	// set url as output parameter
-	if url != "" {
-		action.SetOutput("deployment_url", url)
+func extractURL(action *githubactions.Action, filename string) string {
+	f, err := os.Open(filename)
+	if err != nil {
+		action.Fatalf("%s", err)
 	}
-}
+	defer f.Close()
 
-func getFirstURLFromFile(inputFile *os.File) string {
-	urlPattern := regexp.MustCompile(`(https?://[^\s]+)`)
-	scanner := bufio.NewScanner(inputFile)
+	re := regexp.MustCompile(`(https?://[^\s]+)`)
+	s := bufio.NewScanner(f)
```

Comment:
Go generally uses names that are short for small scopes and longer for larger scopes: https://github.com/golang/go/wiki/CodeReviewComments#variable-names